internal/controller/datadogagent/feature/enabledefault/feature.go

```diff
@@ -101,6 +102,10 @@ type secretInfo struct {
 type clusterAgentConfig struct {
 	serviceAccountName        string
 	serviceAccountAnnotations map[string]string
+
+	resourceMetadataAsTagsClusterRoleName string
+	kubernetesResourcesLabelsAsTags       map[string]map[string]string
+	kubernetesResourcesAnnotationsAsTags  map[string]map[string]string
```

Comment:
`kubernetesResourcesLabelsAsTags` and `kubernetesResourcesAnnotationsAsTags` are not specific to cluster agent, it should be available to other components as well (i.e. node agents).

Reply:
To elaborate on this:

we previously had, for example, `podLabelsAsTags` which is made available to all components. 

On node agents, this implies that the node agent will use this config to generate tags for pods collected directly from the kubelet.

On CLC runners, this implies that the runner will generate pod tags for unschedulable pods that are collected via informers.

On DCA instance, it means the DCA will collect pods from the API server using workloadmeta and will generate tags based on this config.

So in short, all components need to have access to these 2 configuration values, but only the DCA needs the rbacs